### ISSUE TITLE
Headings don't change size for medium/small displays

### ISSUE THREAD

**Issue by ahmetalpbalkan** [Dogfalo/materialize#4229]
I am showing headings in my design and some of them can be 6-7 words long. However they look absolutely terrible because the h1, h2, h3 etc. have exactly the same size on small/medium displays as the large displays.

h1 on large display should not be the same as large display on mobile. You can look at [blog.google](https://blog.google/products/google-wifi/google-wifi-secure-every-point/) for a reference on how to implement this.

#### This is a `h1` on Google Blog:

![image](https://cloud.githubusercontent.com/assets/159209/22868896/a35081c0-f14f-11e6-87d6-2909b8da1d27.png)

#### This is a `h2` on my design yet is' way bigger:

![image](https://cloud.githubusercontent.com/assets/159209/22868901/b4fd8166-f14f-11e6-8107-436974174f35.png)

I think we need more fluid font sizes for headings.

**Comment by kylecordes** [comment]
Closely related - the heading sizes are surprisingly large. Only all the way down at h6 (!!) is there a heading of a suitable size to label sections of UI or content inside something like a <div class=card>. Could the typography make typical semantic headings show up with reasonable style when used inside various materialize constructs?

**Comment by kmmbvnr** [comment]
Last time I checked MaterialIzeCSS implemtation just follows google guides 

https://material.io/guidelines/style/typography.html

**Comment by kmmbvnr** [comment]
Here is the reference implementation

 https://github.com/material-components/material-components-web/blob/master/packages/mdc-typography/_variables.scss

**Comment by kylecordes** [comment]
I brought them up at the same zoom level, and compared side by side, see below. This is just a quick look (I didn't even check the CSS for font sizes...). It does not appear to match.

<img width="1369" alt="typography comparison" src="https://cloud.githubusercontent.com/assets/8799/23828408/e5adb67c-0695-11e7-8f92-011616dee107.png">

**Comment by kmmbvnr** [comment]
At least they are not surprisingly large. That's the specification

**Comment by kylecordes** [comment]
@kmmbvnr Indeed, that is true!

The actual problem I experienced: that there doesn't appear to be a class/style/element in Materialize which corresponds to the "Title" typography from the spec.

**Comment by fega** [comment]
I don't believe that materialize is following the specification, Display 4,3,2,1 are not titles, those should be classes instead of something important in the SEO like the headers.

**Comment by tomscholz** [comment]
Related https://github.com/Dogfalo/materialize/pull/3735
